test/org/ods/usecase/LeVADocumentUseCaseSpec.groovy

```diff
@@ -648,11 +643,10 @@ class LeVADocumentUseCaseSpec extends SpecHelper {
 
         then:
         1 * jiraUseCase.getAutomatedInstallationTestIssues(project.id) >> testIssues
-        1 * jiraUseCase.matchJiraTestIssuesAgainstTestResults(testIssues, testResults, _, _)
+        1 * jiraUseCase.matchJiraTestIssuesAgainstTestResults(testIssues, testResults, _, _) >> null
         1 * usecase.getDocumentMetadata(LeVADocumentUseCase.DOCUMENT_TYPE_NAMES[documentType], project)
         1 * usecase.createDocument(documentType, project, null, _, files, null, null) >> uri
-        1 * usecase.notifyLeVaDocumentTrackingIssue(project.id, documentType, "A new ${LeVADocumentUseCase.DOCUMENT_TYPE_NAMES[documentType]} has been generated and is available at: ${uri}.")
-        1 * jiraUseCase.jira.getIssuesForJQLQuery(jqlQuery) >> [documentIssue]
```

Comment:
@hrcornejo we cannot through these checks away. I wonder why it is working in the other tests and only FTR and IVR fail constantly. Can you help me reproduce the issue on my local machine? According to you, you have been able doing so, but removing the build folder doesn't do the trick for me.

Reply:
I'm out right now but, my environment: 
- Windows 10
- JDK 8 Oracle
- Gradle (wrapper, tested in 5 and 6, depends on commit)
Whit that only this 2 test fail in `JiraUseCase. walkJiraTestIssuesAndTestResults` when try to call:
`this.checkJiraIssueMatchesTestCase(issue, testcase.name)`
The error:
`groovy.lang.MissingMethodException: No signature of method: org.ods.usecase.JiraUseCase.checkJiraIssueMatchesTestCase() is applicable for argument types: (LinkedHashMap, String) values: [[id:1, key:JIRA-1, fields:[summary:1-summary, description:1-description, ...], ...], ...]&#10;Possible solutions: checkJiraIssueMatchesTestCase(java.util.Map, java.lang.String)" type="groovy.lang.MissingMethodException">groovy.lang.MissingMethodException: No signature of method: org.ods.usecase.JiraUseCase.checkJiraIssueMatchesTestCase() is applicable for argument types: (LinkedHashMap, String) values: [[id:1, key:JIRA-1, fields:[summary:1-summary, description:1-description, ...], ...], ...]
Possible solutions: checkJiraIssueMatchesTestCase(java.util.Map, java.lang.String)
at org.ods.usecase.JiraUseCase$_walkJiraTestIssuesAndTestResults_closure18.closure28$_closure29(JiraUseCase.groovy:360)
at org.ods.usecase.JiraUseCase.walkJiraTestIssuesAndTestResults_closure18$_closure28(JiraUseCase.groovy:359)`

I've tried changing this method to (_String_, _String_) but always same error with parameters type, also passing (String String) - (_issue.key.toString()_, _testcase.name_) ¯\_(ツ)_/¯
I guess the closure and mock are the bad guys...

Reply:
How did you manage to reproduce the error locally?

Reply:
Just deleting all the build folder and the first time with gradle test the error comes. The next times (without delete the folder) the errors go away.

Reply:
This is super strange. Because in a `NO_NEXUS=true ./gradlew clean test`, the `clean` actually removes the `build` folder.

Reply:
In my case I had the **NO_NEXUS** like env configured in VSCode. So I only do: gradle test, but the build folder it's not deleted.
And another thing, the second time, all the test (or at least a part of them) are compiled again. I don't know the reason because there is no changes. And this time the mocks works.

Reply:
Fixed the issue by updating the Groovy, Spock and the mock libraries in build.gradle